tests/cbmc/proofs/s2n_socket_read/s2n_socket_read_harness.c

```diff
@@ -0,0 +1,25 @@
+/* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved. */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/**
+ * @file s2n_socket_read_harness.c
+ * @brief Implements the proof harness for s2n_socket_read function.
+ */
+
+#include <utils/s2n_socket.h>
+
+void s2n_socket_read_harness()
+{
+  struct s2n_socket_read_io_context *cbmc_allocate_s2n_socket_read_io_context();
+  /* Non-deterministic inputs. */
+  void *io_context = cbmc_allocate_s2n_socket_read_io_context();
+  const uint8_t *buf = malloc(sizeof(*buf));
+  uint32_t len;
+  
+  /* Operation under verification. */
+  int result = s2n_socket_read(io_context, buf, len);
+
+  /* Post-condition. */
+  assert(S2N_IMPLIES(result >= S2N_SUCCESS, io_context != NULL));
```

Comment:
I don't think using `S2N_SUCCESS` instead of `0` makes sense for this particular harness. The function returns the number of bytes read.

```suggestion
  assert(S2N_IMPLIES(result >= 0, io_context != NULL));
```